streams/test-utils/src/main/java/org/apache/kafka/streams/processor/MockProcessorContext.java

```diff
@@ -96,6 +96,7 @@ public Punctuator getPunctuator() {
             return punctuator;
         }
 
+        @SuppressWarnings("WeakerAccess")
```

Comment:
For my own education: what is this? (btw: can we remove `this` below?)

Reply:
we can remove `this` below.

The WeakerAccess inspection tells you that it's possible to restrict the access scope of `cancel()`. I think this particular case was warning me that `cancel()` could be package-private instead of public. But the static analyzer can only look at the code in the project. We know that we do want the method to be public, so I added a supression for this inspection. An alternative would be to write black-box tests in a different package (just like real user tests would be), and the static analyser wouldn't warn us anymore, since it would have an example of a usage requiring public access.

Reply:
we can remove `this` below.

The WeakerAccess inspection tells you that it's possible to restrict the access scope of `cancel()`. I think this particular case was warning me that `cancel()` could be package-private instead of public. But the static analyzer can only look at the code in the project. We know that we do want the method to be public, so I added a supression for this inspection. An alternative would be to write black-box tests in a different package (just like real user tests would be), and the static analyser wouldn't warn us anymore, since it would have an example of a usage requiring public access.